src/operations.ergo

```diff
@@ -126,13 +126,20 @@ for-file = fn :compiler (Path :file) ^:flags -> {
 
         ##std:doc:value
         ## The object file resulting from compiling $proj-path.
-        object = cache ~key=$key {
+        [:warnings, :object] = cache ~key=$key {
             obj = Path:owned $obj
             fs:create-dir <| Path:parent $obj
-            task ~count=1 "compiling $proj-path" { exec ~env=toolchain:tool-exec-env ^args; () }
-            $obj
+            warnings = task ~count=1 "compiling $proj-path" {
+                exec ~env=toolchain:tool-exec-env ^args |>:stderr | String:from | String:trim
+            }
+            [$warnings, $obj]
         }
 
+        match $warnings [
+            "" -> ()
+            _ -> std:log:warn $warnings
```

Comment:
We should probably change `std:log` to just ignore empty messages. The only use case would be to proxy as a newline, but you should just have newlines in the log message then.